src/Build/Instance/HostServices.cs

```diff
@@ -69,9 +76,30 @@ public ITaskHost GetHostObject(string projectFile, string targetName, string tas
                 return null;
             }
 
-            ITaskHost hostObject = hostObjects.GetAnyMatchingHostObject(targetName, taskName);
+            var monikerNameOrITaskHost =
+                hostObjects.GetAnyMatchingMonikerNameOrITaskHost(targetName, taskName);
 
-            return hostObject;
+            if (monikerNameOrITaskHost == null)
+            {
+                return null;
+            }
+            else
+            {
+                if (monikerNameOrITaskHost.IsMoniker)
+                {
+#if FEATURE_COM_INTEROP
+                    object objectFromRunningObjectTable =
+                        _runningObjectTable.GetObject(monikerNameOrITaskHost.MonikerName);
+                    return (ITaskHost)objectFromRunningObjectTable;
+#else
+                    return null;
```

Comment:
Should this be an error instead? Or is there a situation where we'd expect to get a moniker but not understand it?

Reply:
throwing HostObjectException now